assembl/views/api2/discussion.py

```diff
@@ -743,9 +752,25 @@ def get_visitors(request):
     for st in discussion.agent_status_in_discussion:
         if not getattr(st, attribute, None):
             continue
+        profile_fields = db.query(m.AbstractConfigurableField, m.ProfileField).filter(m.ProfileField.discussion_id == discussion.id).join(m.ProfileField.configurable_field).filter(
```

Comment:
it shouldn't matter in this case, but do an `outerjoin` so that we ensure each row is always present

Reply:
No, why? Keep the join. We have a filter `filter(m.AbstractConfigurableField.identifier == "CUSTOM")`, so even if you do outerjoin (left join), the filter will filter the NULL entries anyway. A ProfileField is required to reference a AbstractConfigurableField.